### DISCUSSION TITLE
isntallation problem

### DISCUSSION THREAD

**Discussion by agosin** [jsingh811/pyAudioProcessing#48] [Q&A] [answered]
Hi, I am having a problem installing the requirements in python3.9 MAC M1, any suggestions?

**Comment by jsingh811** [comment]
> Hi, I am having a problem installing the requirements in python3.9 MAC M1, any suggestions?

Hi @agosin, can you please share the error message that you are observing? Thanks!

**Reply by agosin** [reply]
hi, thank you for reply me.

here is the error:
ModuleNotFoundError: No module named 'pyAudioProcessing'

and then when I try to install the requirements getting several error (attached file)
[terminal.pdf](https://github.com/jsingh811/pyAudioProcessing/files/7542439/terminal.pdf)
)

**Reply by jsingh811** [reply]
Hi @agosin. Thank you for sharing the error. Could you try installing the library as below, and see if that resolves the problem?  Thanks!
`
python -m pip install pyAudioProcessing
`

**Comment by agosin** [comment]
Hi, same error, it is attached

El mar, 16 nov 2021 a las 16:19, Jyotika Singh ***@***.***>)
escribió:

> Hi @agosin <https://github.com/agosin>. Thank you for sharing the error.
> Could you try installing the library as below, and see if that resolves the
> problem? Thanks!
> python -m pip install pyAudioProcessing
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/jsingh811/pyAudioProcessing/discussions/48#discussioncomment-1654925>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAIKYYJVEU3RC6N3GA2LQI3UMLYJFANCNFSM5HFY3IWQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
Esteban Agosin Otero
www.estebanagosin.cl
+1 (253)290-7068

**Comment by agosin** [comment]
it is here the file
[terminal2.pdf](https://github.com/jsingh811/pyAudioProcessing/files/7551753/terminal2.pdf)

**Reply by jsingh811** [reply]
Hi @agosin. It appears to be a problem reported with macOS 11. https://github.com/matplotlib/matplotlib/issues/18678
As a workaround, I recommend trying the below method for getting this library installed.
```
git clone git@github.com:jsingh811/pyAudioProcessing.git
cd pyAudioProcessing
```
This will create a pyAudioProcessing directory in your cwd.
Open the file pyAudioProcessing/requirements/requirements.txt and comment out the 2nd line like `#matplotlib==3.3.4` , save and close the file.
Then run the below in your terminal
```
pip install -e .
```
Please let me know if there are any further errors.

**Comment by jsingh811** [comment, marked as answer]
Hi @agosin, I was able to test installation on M1 and it worked fine. I was using anaconda distribution of Python instead. Please open a new terminal session after [download anaconda](https://docs.anaconda.com/anaconda/install/mac-os/) and completing installation steps. Verify `which python` output is now anaconda3 path, & `pip install pyAudioProcessing==1.1.9` should then work fine. Please let me know if issues persist. Thanks.

**Comment by agosin** [comment]
Hi, sorry for the late reply, I was out of the office. got it! now starting
to understand the code. but no error in installation.


El vie, 7 ene 2022 a las 15:36, Jyotika Singh ***@***.***>)
escribió:

> Hi @agosin <https://github.com/agosin>, I was able to test installation
> on M1 and it worked fine. I was using anaconda distribution of Python
> instead. Please open a new terminal session after download anaconda
> <https://docs.anaconda.com/anaconda/install/mac-os/> and completing
> installation steps. Verify which python output is now anaconda3 path, & pip
> install pyAudioProcessing==1.1.9 should then work fine. Please let me
> know if issues persist. Thanks.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/jsingh811/pyAudioProcessing/discussions/48#discussioncomment-1929595>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAIKYYMAXOMOWUHSBAO2RB3UU52HTANCNFSM5HFY3IWQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***
> com>
>


-- 
Esteban Agosin Otero
www.estebanagosin.cl
+1 (253)290-7068
